python/user_roles/update_roles.py

```diff
@@ -0,0 +1,140 @@
+        return self.role_ids
+        
+# Retuns a dictionary of user's email to user object.
+def get_user_ids(base_url, headers):
+    users = {}
+    list_users_url = f"{base_url}users"
+
+    response = requests.get(list_users_url, headers=headers)
+    if response.status_code != 200:
+        print(f"List Users Error: {response.status_code} with {list_users_url}")
+        sys.exit(1)
+    
+    resp_json = response.json()
+    users_resp = resp_json['users']
+
+    for user in users_resp:
+        #print(f"{user['email']} : {user['id']}")
+        a_user = User(user)
+        users[user['email']] = a_user
+    
+    return users
+
+# Returns a dictionary of role names to role ID.
+def get_role_ids(base_url, headers):
+    roles = {}
+    list_roles_url = f"{base_url}roles"
+
+    response = requests.get(list_roles_url, headers=headers)
+    if response.status_code != 200:
+        print(f"List Roles Error: {response.status_code} with {list_roles_url}")
+        sys.exit(1)
+    
+    resp_json = response.json()
+    roles_resp = resp_json['roles']
+
+    for role in roles_resp:
+        #print(f"{role['name']} : {role['id']}")
+        roles[role['name']] = role['id']
+    
+    return roles
+
+# Returns the role IDs for a list of role_name using the role_name_to_id dictionary.
+def map_role_names_to_ids(role_name_to_id, role_names):
+    role_ids = []
+    role_names = [role.strip(' ') for role in role_names]
+
+    for role_name in role_names:
+        role_id = role_name_to_id.get(role_name)
+        if not role_id:
+            print(f"{role_name} is not on system")
+            continue
+        role_ids.append(role_id)
+
+    role_ids.sort()
+    return role_ids
+
+# Updates the user with new roles.
+def update_user(base_url, headers, user_id, role_ids_to_update):
+    update_user_url = f"{base_url}users/{user_id}"
+
+    update_params = {
+        "user": {
+            "role_ids": role_ids_to_update
+        }
+    }
+
+    print(f"Updating: {user_id} - {role_ids_to_update}")
+
+    response = requests.put(update_user_url, headers=headers, data=json.dumps(update_params))
+    if response.status_code != 204:
+        print(f"Update User Error: {response.status_code} with {update_user_url}")
+        return
+    
+
+if __name__ == "__main__":
+    print("Update User Roles")
+    print("")
+
+    csv_file_name = "user_roles.csv"
+    if len(sys.argv) > 1:
+        csv_file_name = sys.argv[1]
+
+    # Obtain the Kenna Security API key from an environment variable.
+    api_key = os.getenv('KENNA_API_KEY')
+    if api_key is None:
+        print("API key is non-existent")
+        sys.exit(1)
+    
+    # HTTP headers.
+    headers = {'X-Risk-Token': api_key,
+               'Content-Type': 'application/json; charset=utf-8',
+               'User-Agent': 'sample.connector_ui/1.0.0 (Cisco Secure)'}
+    
+    # You might have to change this depending on your deployment.
+    base_url = "https://api.kennasecurity.com/"
```

Comment:
optional thing – not necessary – but we could make this the default base_url, but overrideable via an optional argument or environment variable.

Reply:
This is a good idea.  I will add it to the next blog.